src/lib/utilities.js

```diff
@@ -44,3 +45,64 @@ export function uuid() {
     s4() + '-' + s4() + s4() + s4();
 }
 
+
+// The following is all for global error handling.
+// We keep the globalError value on top of AppController's 
+// this.state.error because this.setState is asynchronous
+// while this variable will be synchronous and can therefore
+// always be checked
+let globalError = null;
+let appControllerMounted = false;
```

Comment:
setAppReady already tracks if appController is mounted. just import getAppReady from falcorUtils.